### ISSUE TITLE
Atom 1.18.0 Beta Linux CI Travis build fails. GLIBC not found.

### ISSUE THREAD

**Issue by gandm** [atom/atom#14440]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Atom 1.18.0 Beta Linux CI Travis build fails. GLIBC not found.

### Steps to Reproduce

See https://travis-ci.org/gandm/language-babel/jobs/233108682

**Expected behavior:** 

To Run & pass.

**Actual behavior:** [What actually happens]

Doesn't run

**Reproduces how often:** [What percentage of the time does it reproduce?]

100%

### Versions

Atom    : 1.18.0-beta0
Electron: 1.3.15
Chrome  : 52.0.2743.82
Node    : 6.5.0
Using APM version: ???

**Comment by lgeiger** [comment]
We ran into the same problem.
Switching to the Travis `trusty` image fixed it for us: https://github.com/nteract/hydrogen/pull/784

Looks like the new GitHub package needs a newer version of `libc`.

**Comment by BinaryMuse** [comment]
The trace is coming from `git-utils`, which is not new AFAIK. @smashwilson, does this make sense to you?

**Comment by Arcanemagus** [comment]
For the sake of searching, the full text of the error is:
```
module.js:597
  return process.dlopen(module, path._makeLong(filename));
                 ^
Error: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.17' not found (required by /home/travis/atom/usr/share/atom-beta/resources/app/apm/node_modules/git-utils/build/Release/git.node)
    at Error (native)
    at Object.Module._extensions..node (module.js:597:18)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/travis/atom/usr/share/atom-beta/resources/app/apm/node_modules/git-utils/lib/git.js:8:16)
    at Object.<anonymous> (/home/travis/atom/usr/share/atom-beta/resources/app/apm/node_modules/git-utils/lib/git.js:371:4)
    at Module._compile (module.js:570:32
```

Also it appears that this is being printed whenever `apm` is called, so this may be an issue in APM not Atom directly.

**Comment by BinaryMuse** [comment]
@Arcanemagus pointed me at a commit that reminded me what caused this — we updated Keytar in the latest version of apm to use libsecret to work with a wider array of providers. This did indeed require that we update to trusty image on Travis.

**Comment by smashwilson** [comment]
Yep! When we upgraded our Travis build to use Trusty in 8838b62737783, native modules began to build against [glibc 2.19 rather than 2.15](https://en.wikipedia.org/wiki/GNU_C_Library#Version_history). We addressed this somewhat for dugite for the github package side in desktop/dugite-native#20, but that was a very git-specific workaround which I'm still not convinced was a good idea.

We can do some combination of:

* _(short-term solution)_ Ask packages to upgrade their builds to use `dist: trusty`, too. To facilitate this we could (1) update the documentation to reflect this and (2) improve the test script to detect older libc versions and suggest this solution.
* _(medium-term solution)_ A little `ldd` sleuthing to see which newer libc call(s) in git-utils are being picked up as GLIBC_2.17 and see if we can rewrite to avoid them;
* _(long-term solution)_ Avoid the native dependency entirely by rewriting git-utils in terms of a git operation service exposed by the github package.

In general, it may also be nice to augment the build with some kind of automated probing for the minimum required libc version we inherit from all of our bundled `.node` files so that we aren't caught off-guard by this as we upgrade both Travis and our native deps.
